.github/workflows/ci.yml

```diff
@@ -14,7 +14,7 @@ jobs:
       fail-fast: true
       matrix:
         os: [ubuntu-latest, macOS-latest]
-        ghc: ['9.2', '9.0', '8.10', '8.8', '8.6', '8.4', '8.2']
```

Comment:
I suspect you need to specify `9.4.2` to make it work.

Reply:
Not sure. The issue appears to be GHC 9.4(.2) is already installed in the MacOS environment, and `ghcup` doesn't want to proceed unless `--force` is used. However, since `ghcup` is invoked by `haskell/actions/setup`, I need to figure out how to pass things through.

Reply:
No, this is a bug in ghcup and haskell actions uses an ancient version.

I suggest to drop haskell setup action and just use straight ghcup with a manual matrix.

Reply:
Is it an option to use HaskellCI (as was done for TravisCI in the past, looking at some leftover files in the repo) to set up all Linux jobs?

Reply:
Sure